### PR TITLE
Branch cost for adding and editing price of grocery

### PR DESCRIPTION
**Pull request by LuoYu-uwu**

### CODE REVIEW

File: src/main/java/exceptions/InvalidCostException.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package exceptions;
+
+/**
+ * Represents the exception thrown when the cost inputted by the user is invalid.
+ */
+public class InvalidCostException extends GitException{
```
Comment (64-1) [inline src/main/java/exceptions/InvalidCostException.java:6]: The exception class InvalidCostException has been added properly. It extends GitException, which fits the design of having a hierarchy of exceptions in the application. The message "Please input a valid cost!" is clear and provides straightforward feedback to the user.

File: src/main/java/grocery/Grocery.java [hunk 2]
```diff
@@ -18,10 +19,11 @@ public class Grocery {
      * @param amount Measurement of grocery.
      * @param expiration When grocery expires.
      */
-    public Grocery(String name, int amount, LocalDate expiration) {
+    public Grocery(String name, int amount, LocalDate expiration, double cost) {
```
Comment (64-1) [inline src/main/java/grocery/Grocery.java:22]: The addition of the cost attribute and corresponding methods (setCost and getCost) are implemented correctly. The printGrocery method has been updated to include cost information, which aligns with the feature's requirements.
Consider validating the cost in the setCost method to ensure it's a positive value, as negative costs might not be meaningful in this context.

**Review by 64-1 — APPROVED** [review]

File: src/main/java/git/Ui.java [hunk 3]
```diff
@@ -73,6 +78,8 @@ public String[] processInput() {
 
     /**
      * Prompts user for expiration date.
+     *
+     * @return the formatted expiration date
```
Comment (wallywallywally) [inline src/main/java/git/Ui.java:82]: I think the first word after @return should be capitalised as stated in the se-edu guides! Same for other parameters like @param

**Review by wallywallywally — COMMENTED** [review]
LGTM, just some issues with capitalisation in the Javadoc comments!

**Pull request merged**